learntools/embeddings/nbs/1-exercises.ipynb

```diff
@@ -117,6 +117,9 @@
     "\n",
     "**TODO: diagram**\n",
     "\n",
+    "#### DB: I agree in general that we should keep negative results, and ideally use them as food for thought to users (why are they negative). I wouldn't have expected this to make much difference in this case. To the extent a bias term is useful, it should automatically be included in the embedding.  So adding a bias seems to be weakly dominated (modulo the issue of overfitting) by using an embedding size of 9, and we wouldn't have expected that to make a big difference either. Unless you think I'm missing something with this logic, I'd limit work on bias to a single short question.\n",
```

Comment:
> To the extent a bias term is useful, it should automatically be included in the embedding.  So adding a bias seems to be weakly dominated (modulo the issue of overfitting) by using an embedding size of 9, and we wouldn't have expected that to make a big difference either.

You kind of touched on this, but the only thing I'd add here is that this is sort of a form of regularization - we're imposing some prior that says each movie has some intrinsic level of goodness/badness that's orthogonal to personal taste. (Or, in the case of user biases, a prior that each user has some degree of generosity or harshness to their ratings). If that reflects the reality of the data, then I could see it helping convergence/generalization.

In the case of the matrix factorization architecture, yeah, it's basically equivalent to increasing the embedding size by one, but also forcing all user embeddings to have a value of 1.0 in the new dimension.

In the case of the DNN architecture that we're using in this first lesson, I think it's more interesting, because it introduces a direct connection between the input layer and the output, which skips over the hidden layers in between (particularly the penultimate bottleneck layer).